components/ai-services-mgt/org.wso2.carbon.ai.service.mgt/src/main/java/org/wso2/carbon/ai/service/mgt/token/AIAccessTokenManager.java

```diff
@@ -0,0 +1,265 @@
+                "AIServices.LoginFlow.TokenRequestMaxRetries") != null ?
+                Integer.parseInt(IdentityUtil.getProperty("AIServices.LoginFlow.TokenRequestMaxRetries")) : 3;
+        private static final long TIMEOUT = IdentityUtil.getProperty(
+                "AIServices.LoginFlow.TokenRequestTimeout") != null ?
+                Long.parseLong(IdentityUtil.getProperty("AIServices.LoginFlow.TokenRequestTimeout")) : 3000;
+
+        AccessTokenRequestHelper(String key, String tokenEndpoint, CloseableHttpAsyncClient client) {
+
+            this.client = client;
+            this.gson = new GsonBuilder().create();
+            this.key = key;
+            this.aiServiceTokenEndpoint = tokenEndpoint;
+        }
+
+        /**
+         * Request access token to access the Login Flow AI service.
+         *
+         * @return the JWT access token.
+         * @throws AIServerException If an error occurs while requesting the access token.
+         */
+        public String requestAccessToken() throws AIServerException {
+
+            String tenantDomain = getTenantDomainFromContext();
+            LOG.info("Initiating access token request for Login Flow AI service from tenant: " + tenantDomain);
+            try {
+                client.start();
+                for (int attempt = 0; attempt < MAX_RETRIES; attempt++) {
+                    HttpPost post = new HttpPost(aiServiceTokenEndpoint);
+                    post.setHeader("Authorization", "Basic " + key);
+                    post.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+                    StringEntity entity = new StringEntity("grant_type=client_credentials");
+                    entity.setContentType(new BasicHeader(HTTP.CONTENT_TYPE, "application/x-www-form-urlencoded"));
+                    post.setEntity(entity);
+
+                    CountDownLatch latch = new CountDownLatch(1);
+                    final String[] accessToken = new String[1];
+                    client.execute(post, new FutureCallback<HttpResponse>() {
+                        @Override
+                        public void completed(HttpResponse response) {
+
+                            try {
+                                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                                    String responseBody = EntityUtils.toString(response.getEntity());
+                                    Map<String, Object> responseMap = gson.fromJson(responseBody, Map.class);
+                                    accessToken[0] = (String) responseMap.get("access_token");
+
+                                    // Decode the JWT to extract client ID.
+                                    String[] jwtParts = accessToken[0].split("\\.");
+                                    if (jwtParts.length == 3) {
+                                        String payloadJson = new String(Base64.getUrlDecoder().decode(jwtParts[1]),
```

Comment:
Check if we need to decode the token also check if we actually need the token type